--- /app/share/vala/vapi/gee-1.0.vapi	1970-01-01 01:00:00.000000000 +0100
+++ gee-1.0.vapi	2018-07-21 11:10:36.402468624 +0200
@@ -325,9 +325,6 @@
 			public abstract V value { get; set; }
 		}
 		public abstract void clear ();
-		[Deprecated]
-		public abstract bool contains (K key);
-		[Deprecated]
 		public abstract bool contains_all (Gee.Map<K,V> map);
 		public static Gee.Map<K,V> empty<K,V> ();
 		public abstract V @get (K key);
@@ -335,10 +332,6 @@
 		public abstract bool has_all (Gee.Map<K,V> map);
 		public abstract bool has_key (K key);
 		public abstract Gee.MapIterator<K,V> map_iterator ();
-		[Deprecated]
-		public abstract bool remove (K key, out V value = null);
-		[Deprecated]
-		public abstract bool remove_all (Gee.Map<K,V> map);
 		public abstract void @set (K key, V value);
 		public abstract void set_all (Gee.Map<K,V> map);
 		public abstract bool unset (K key, out V value = null);
