--- /usr/share/vala-0.36/vapi/glib-2.0.vapi	1970-01-01 01:00:00.000000000 +0100
+++ build/glib-2.0.vapi	2017-08-10 22:40:36.846872261 +0200
@@ -4581,7 +4581,7 @@
 		[Version (since = "2.4")]
 		public int index (G data);
 		[Version (since = "2.4")]
-		public void remove (G data);
+		public bool remove (G data);
 		[Version (since = "2.4")]
 		public void remove_all (G data);
 		[Version (since = "2.4")]
