--- /app/share/vala/vapi/unity.vapi	2020-05-10 18:13:09.748464995 +0200
+++ build/unity.vapi	2020-05-10 22:31:02.304959076 +0200
@@ -94,8 +94,6 @@
 		public void clear ();
 		public unowned string[]? get_categories (string id);
 		public static Unity.AppInfoManager get_default ();
-		[Deprecated (replacement = "AppInfoManager.get_default")]
-		public static Unity.AppInfoManager get_instance ();
 		public unowned string[]? get_keywords (string id);
 		public string? get_path (string id);
 		public GLib.AppInfo? lookup (string id);
