--- /usr/share/vala-0.38/vapi/webkit2gtk-web-extension-4.0.vapi	1970-01-01 01:00:00.000000000 +0100
+++ vapi/webkit2gtk-web-extension-4.0.vapi	2017-08-09 21:37:18.294369745 +0200
@@ -2599,10 +2599,9 @@
 		[CCode (cheader_filename = "webkit2/webkit-web-extension.h", type_id = "webkit_dom_event_target_get_type ()")]
 		[GIR (name = "DOMEventTarget")]
 		public interface EventTarget : GLib.Object {
-			public bool add_event_listener_with_closure (string event_name, [CCode (type = "GClosure*")] owned WebKit.DOM.EventTargetFunc handler, bool use_capture);
+			public bool add_event_listener(string event_name, [CCode (delegate_target_pos = -1)] EventTargetFunc handler, bool use_capture);
 			public abstract bool dispatch_event (WebKit.DOM.Event event) throws GLib.Error;
-			public abstract bool remove_event_listener (string event_name, GLib.Closure handler, bool use_capture);
-			public bool remove_event_listener_with_closure (string event_name, GLib.Closure handler, bool use_capture);
+			public abstract bool remove_event_listener (string event_name, EventTargetStaticFunc handler, bool use_capture);
 		}
 		[CCode (cheader_filename = "webkit2/webkit-web-extension.h", type_id = "webkit_dom_node_filter_get_type ()")]
 		[GIR (name = "DOMNodeFilter")]
@@ -2616,6 +2615,8 @@
 		}
 		[CCode (cheader_filename = "webkit2/webkit-web-extension.h")]
 		public delegate void EventTargetFunc (WebKit.DOM.EventTarget target, WebKit.DOM.Event event);
+		[CCode (cheader_filename = "webkit2/webkit-web-extension.h", has_target=false)]
+		public delegate void EventTargetStaticFunc (void* data, WebKit.DOM.EventTarget target, WebKit.DOM.Event event);
 		[CCode (cheader_filename = "webkit2/webkit-web-extension.h", cname = "WEBKIT_DOM_CSS_RULE_CHARSET_RULE")]
 		public const int _CSS_RULE_CHARSET_RULE;
 		[CCode (cheader_filename = "webkit2/webkit-web-extension.h", cname = "WEBKIT_DOM_CSS_RULE_FONT_FACE_RULE")]
